Cabal/src/Distribution/Simple/Build.hs

```diff
@@ -146,6 +158,15 @@ build pkg_descr lbi flags suffixes = do
   verbosity = fromFlag (buildVerbosity flags)
 
 
+checkBuildProblems
+  :: Verbosity -> Compiler -> BuildFlags -> IO ()
+checkBuildProblems verbosity comp flags = do
+    unless (jsemSupported comp || not (isJust (flagToMaybe (buildUseSemaphore flags))) ) $
+        die' verbosity $
+              "Your compiler does not support the -jsem flag. "
+           ++ "To use this feature you must use GHC 9.7 or later."
```

Comment:
Should this say 9.8 for consistency with what the changelog and documentation (parts of this same PR) say?

Reply:
Done.